rfcs/0001-fobnail-architecture-change.md

```diff
@@ -0,0 +1,143 @@
+# Summary
+
+Change Fobnail architecture - instead of having server on the attester and
+client on Fobnail Token, server would be located on Fobnail Token and client on
+Attester.
+
+# Motivation
+
+The current architecture severely limits Fobnail Token configuration abilities
+due to Fobnail Token being a client. Fobnail Token has hardcoded paths for
+performing provisioning and attestation. We are not capable of sending requests
+to Fobnail Token which severely limits the amount configuration possible, e.g.
+an once provisioned platform can not be unprovisioned without resetting Fobnail
+Token into factory state. Fobnail Token configuration cannot be changed without
+resetting and going once again through the provisioning process.
+
+It is also not possible to expose any services from Fobnail Token to Attester,
+such as access to symmetric cryptographic keys or performing cryptographic
+operations on behalf of attester in case of asymmetric cryptography. This is a
+must for features like disk encryption or Fobnail-based authentication to
+services, like VPN.
+
+Switching to a server based Fobnail Token allows to expose services for
+accessing cryptographic keys and performing administrative tasks.
+
+# Architecture change proposal
+
+## Fobnail Token provisioning
+
+Currently, if in unprovisioned state, Fobnail Token initiates provisioning by
+requesting PO certificate chain from Platform Owner, then it verifies PO
+certificate chain against a trusted certificate embedded into Fobnail firmware.
+If verification passes, Fobnail generates Identity/Encryption keypair and
+requests PO to generate certificate from this key. Retrieval of valid
+Identity/Encryption certificate finalizes provisioning process.
+
+Instead we propose the following protocol:
+1. Unprovisioned Fobnail token awaits for server to commence provisioning by
+   sending it's certificate chain.
+2. Fobnail verifies certificate chain against trusted certificate embedded into
+   its firmware. If verification is succesful, Fobnail generates
+   Identity/Encryption keypair and responds with Certificate Signing Request.
+   Otherwise an error is returned.
+3. At this point Platform Owner configures Fobnail, like default appraisal
+   policy. This step is optional, if skipped, defaults will be used.
+4. Platform finalizes configuration by sending certificate generated from
+   previously provided CSR.
```

Comment:
Platform or server? I'm not sure how those terms are used here. What Platform means here so it can do anything, I thought it is hardware.

Reply:
Should be Platform Owner, fixed.